pkg/network/tracer/tracer_test.go

```diff
@@ -2071,3 +2072,52 @@ func TestSelfConnect(t *testing.T) {
 	}, 5*time.Second, time.Second, "could not find expected number of tcp connections, expected: 1")
 
 }
+
+func TestNewConntracker(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	cfg := testConfig()
+
+	mockCreator := func(_ *config.Config) (netlink.Conntracker, error) {
+		return netlink.NewMockConntracker(ctrl), nil
+	}
+
+	errCreator := func(_ *config.Config) (netlink.Conntracker, error) {
+		return nil, assert.AnError
+	}
+
+	mockConntracker := netlink.NewMockConntracker(ctrl)
+	noopConntracker := netlink.NewNoOpConntracker()
```

Comment:
using a mock seems like overkill here. You could do something like:

```
createdConntracker := &noopConntracker{}
mockCreator := func(_ *config.Config) (netlink.Conntracker, error) {
		return netlink.NewMockConntracker(ctrl), nil
	}
```

and check for pointer equality later (instead of the checking the type) in the test

Reply:
It complicates the check later for type, since we still have to check for type for the actual noop case.